hr_timesheet_hours_billed/models/hours_billed.py

```diff
@@ -0,0 +1,45 @@
+from odoo import api, fields, models
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    approved = fields.Selection(
+        selection=[("no", "No"), ("yes", "Yes")],
+        default="no",
+        required=True,
+    )
+    approved_uid = fields.Many2one(
+        "res.users",
+        string="Approved by",
+        index=True,
+        tracking=True,
+        compute="_compute_approved_uid",
+        store=True,
+    )
+    approved_date = fields.Date(
+        copy=False, compute="_compute_approved_date", store=True
```

Comment:
You can use a singe compute method for both fields. They are always updated simultaneously